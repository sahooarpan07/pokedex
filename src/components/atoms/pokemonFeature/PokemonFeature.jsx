import React from 'react';
import {View, Text} from 'react-native';

const PokemonFeature = ({header, value, fullWidth}) => {
  return (
    <View
      style={{
        flexDirection: 'column',
        width: fullWidth ? '100%' : '50%',
        marginTop: 16,
      }}>
      <Text
        style={{
          fontWeight: 'bold',
          color: '#180345',
          fontSize: 20,
        }}>
        {header}
      </Text>
      <Text style={{marginTop: 20}}>{value}</Text>
    </View>
  );
};

export default PokemonFeature;
