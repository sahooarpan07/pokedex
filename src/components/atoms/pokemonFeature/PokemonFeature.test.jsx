/**
 * @format
 */

import 'react-native';
import React from 'react';
import PokemonFeature from './PokemonFeature';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('should render Pokemon Feature', () => {
  const tree = renderer
    .create(<PokemonFeature header={'Weight'} value={'68 kg'} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
