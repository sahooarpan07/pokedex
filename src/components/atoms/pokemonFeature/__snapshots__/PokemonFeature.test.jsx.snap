// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pokemon Feature 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "marginTop": 16,
      "width": "50%",
    }
  }
>
  <Text
    style={
      {
        "color": "#180345",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Weight
  </Text>
  <Text
    style={
      {
        "marginTop": 20,
      }
    }
  >
    68 kg
  </Text>
</View>
`;
